addons/web/static/tests/views/view_dialogs_tests.js

```diff
@@ -184,7 +184,7 @@ QUnit.module('Views', {
         });
 
         var dialog;
-        new dialogs.SelectCreateDialog(parent, {
+        await new dialogs.SelectCreateDialog(parent, {
```

Comment:
this shouldn't be necessary

Reply:
It's the only way I found to fix the build issue: the `open` call is async (because it starts with widget) but we were not waiting for the result, it failed locally and on runbot, adding an await fixes the problem.

Reply:
You can mock the session by providing it in params of createParent:
```js
const parent = await createParent({
    data: this.data,
    /* ... */
    session: {
        user_has_group: (() => Promise.resolve(false)), // prevent from doing an rpc
    },
});
```

Reply:
you can also remove the unused variable "dialog", and the `.then()` by the way

Reply:
> You can mock the session by providing it in params of createParent:

That seems like a strictly workse workaround though. Since opening the dialog is async, surely it should be awaited and it's a latent bug not to do so? Though if I'm going to clean up more stuff I can probably also remove the nextTick await, it's basically failable reliance on an implementation detail (at best) which should now be unnecessary,

Reply:
In a mocked environment where no rpcs are done, waiting for a nextTick() is (almost) always enough. I personally prefer that over await the opening of the dialog, as this could hide real rpcs.

Reply:
> as this could hide real rpcs.

If they're undesirable (which seems to be what you imply) then surely "real RPCs" should fail clearly and noisily unless explicitly allowed, no?

Reply:
@aab-odoo session mock thing seems to not work (see build)

Reply:
We could indeed make them automatically fail, but this is a mess. Not sure overriding `session.rpc` would be enough. We could give it a try though (e.g. in `_mockGlobalObjects`, test_utils_mock.js).

Reply:
> @aab-odoo session mock thing seems to not work (see build)

Indeed, rpc is still done:
`2020-07-07 10:51:11,832 15 INFO 3485672-54114-45412d-all werkzeug: 127.0.0.1 - - [07/Jul/2020 10:51:11] "POST /web/dataset/call_kw/res.users/has_group HTTP/1.1" 200 - 7 0.001 0.003`

Reply:
`SelectCreateDialog list view in readonly` it's not the same test

Reply:
> `SelectCreateDialog list view in readonly` it's not the same test

No but since it's a test which didn't previously fail and which I didn't touch, I'd expect it's because of a promise bleed, something along those lines.

Reply:
Don't forget that the suite is run in failfast mode, so a failing test might hide a forest of failing tests

Reply:
> Don't forget that the suite is run in failfast mode, so a failing test might hide a forest of failing tests

Right but nothing failed previously. The breakage here is mocking the session in a different test (re-adding the await did not fix it, removing the session mock did).

Reply:
To sum up:

1. the base tests fails with

    ```
    "QUnit test failed: "Views > view_dialogs > SelectCreateDialog use domain, group_by and search default" [message: "Promise rejected during "SelectCreateDialog use domain, group_by and search default": Found 0 nodes instead of 1.", expected: "undefined"]
    "QUnit test failed: "Views > view_dialogs > SelectCreateDialog use domain, group_by and search default" [message: "Expected 3 assertions, but 2 were run", expected: "undefined"]
    ```
2. awaiting the SelectCreateDialog fixes it
3. mocking the session (in that specific test) breaks a bunch of other stuff, regardless of [2], tried mocking user_has_group, uid and nothing (just mocking session with an empty object) and they all break one of the following tests

Reply:
Anyway I reset the PR to the iteration which worked.

Reply:
@xmo-odoo The other tests pass because the rpc is done by the first one (the one where you added `await`), as there is a cache in the session. If we mock it in the first test, the second one that needs the same information would do the rpc, and thus crash. This is obviously not acceptable (tests have side-effects, running them in a different order may change the outcome...). With the new rpc you added in list controller, a lot of tests should be adapted to mock the corresponding request (being with mockRPC, or with user_has_group directly). This won't be convenient. As this rpc is done in a major element of the webclient, all the time, I'd suggest to mock it automatically (to return false), in `mockGlobalObjects` (`addMockEnvironment` in 13.3), like a "default" mock to use if it is not specifically overridden for the given test.

Reply:
Makes sense, and explains the odd behaviour.